### PR TITLE
fix: pull request page bugfixes

### PR DESCRIPTION
**Pull request by shaurya-harness**
fixes : 
1. default branch as the initial selected branch when creating PR
2. PR restore branch / delete branch button after PR merge
3. Diff View Switch Unified/Split
4. PR Panel auto expand when nothing to show more fixed 

Screenshot 2024-11-21 at 2 22 51 PM

Screenshot 2024-11-21 at 2 22 58 PM

Screenshot 2024-11-21 at 2 22 33 PM

### CODE REVIEW

File: apps/gitness/package.json [hunk 1]
```diff
@@ -46,6 +46,7 @@
   },
   "devDependencies": {
     "@eslint/js": "^9.9.0",
+    "@git-diff-view/react": "^0.0.16",
```
Comment (praneshg239) [inline apps/gitness/package.json:49]: please raise a SECOPS ticket to review this package.

Comment (shaurya-harness) [reply apps/gitness/package.json:49]: it is already approved and present in views package as well

File: apps/gitness/src/pages/pull-request/pull-request-compare-page.tsx [hunk 2]
```diff
@@ -141,6 +141,14 @@ export const CreatePullRequest = () => {
     memorizedState
   ])
   const { data: { body: repoMetadata } = {} } = useFindRepositoryQuery({ repo_ref: repoRef })
+
+  useEffect(() => {
+    if ((!diffTargetBranch || !diffSourceBranch) && repoMetadata?.default_branch) {
+      setSelectedTargetBranch(diffTargetBranch ?? repoMetadata?.default_branch)
+      setSelectedSourceBranch(diffSourceBranch ?? repoMetadata?.default_branch)
+    }
+  }, [repoMetadata])
```
Comment (praneshg239) [inline apps/gitness/src/pages/pull-request/pull-request-compare-page.tsx:150]: do we need to add `diffTargetBranch` and `diffSourceBranch` to effect deps?

Comment (shaurya-harness) [reply apps/gitness/src/pages/pull-request/pull-request-compare-page.tsx:150]: no, they are either always present in params when you move to a PR creation page via new commit, or never present if you open a create new PR page

**Review by praneshg239 — APPROVED** [review]

**Review by cjlee01 — APPROVED** [review]

**Pull request merged**